### ISSUE TITLE
[Bug]: user-select: none is being applied in multiple components in v9

### ISSUE THREAD

**Issue by khmakoto** [microsoft/fluentui#23589]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Reproduction

N/A

### Bug Description

Multiple components in `@fluentui/react-components` are using `user-select: 'none'`, which is a bad pattern for accessibility because it makes text impossible to copy and prevents people from using tools like read-aloud or dictionary lookup. The following is the list of components that has this set, and which should be evaluated on a case by case basis:
- [x] `AccordionHeader` - #23713
- [x] ~`Avatar`~ - We actually want this style in this component 
- [x] `Checkbox` - #23590
- [x] ~`MenuItem`~ - We actually want this style in this component 
- [x] `Radio` - #23590
- [x] `Slider` - #23590
- [x] `Switch` - #23590

### Requested priority

Normal

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
